stwcs/wcsutil/altwcs.py

```diff
@@ -490,6 +460,88 @@ def _getheader(fobj, ext):
     return hdr
 
 
+def wcs_from_key(fobj, ext, from_key=' ', to_key=None):
+    """
+    Read in WCS with a given ``from_key`` from the specified extension of
+    the input file object and return a FITS header representing this WCS
+    using desired WCS key specified by ``to_key``.
+
+    Parameters
+    ----------
+    fobj : str, `astropy.io.fits.HDUList`
+        FITS filename or `astropy.io.fits.HDUList` object containing
+        a header with an alternate/primary WCS to be read.
+
+    ext : int, str or tuple of (str, int)
+        Extension specification identifying image HDU from which WCS should be
+        loaded. If ``ext`` is a tuple, it is of the form ``(extname, extver)``
+        where ``extname`` is a `str` extension name and ``extver`` is
+        an integer extension version.
+        An integer ``ext`` indicates "extension number". Finally, a single
+        `str` extension name is interpretted as ``(ext, 1)``.
+
+    from_key : {' ', 'A'-'Z'}
+        A 1 character string that is either a space character indicating the
+        primary WCS, or one of the 26 ASCII letters (``'A'``-``'Z'``)
+        indicating alternate WCS to be loaded from specified header.
+
+    to_key : {None, ' ', 'A'-'Z'}
+        The key of the primary/alternate WCS to be used in the returned header.
+        When ``to_key`` is `None`, the returned header describes a WCS with the
+        same key as the one read in using ``from_key``. A space character or
+        a single ASCII letter indicates the key to be used for the returned
+        WCS (see ``from_key`` for details) and it **must** be different from
+        ``from_key`` parameter.
+
+    Returns
+    -------
+    hdr: astropy.io.fits.Header
+        Header object with FITS representation for specified primary or
+        alternate WCS.
+
+    """
+    if len(from_key) != 1 or from_key.strip() not in string.ascii_uppercase:
+        raise ValueError(
+            "Parameter 'from_key' must be a character - one of 'A'-'Z' or ' '."
+        )
+
+    if to_key is None:
+        to_key = from_key
+
+    elif len(to_key) != 1 or to_key.strip() not in string.ascii_uppercase:
+        raise ValueError(
+            "Parameter 'to_key' must be a character - one of 'A'-'Z' or ' '."
+        )
+
+    elif to_key == from_key:
+        raise ValueError(
+            "When parameter 'to_key' is not None, it must be different from 'from_key'."
+        )
```

Comment:
On second thought it doesn't make sense to raise an error. If someone sets it it was a conscious decision. Please remove it.